src/java.base/share/classes/sun/security/ssl/SSLConfiguration.java

```diff
@@ -133,6 +139,25 @@ final class SSLConfiguration implements Cloneable {
         useExtendedMasterSecret = supportExtendedMasterSecret;
     }
 
+    /*
+     * If the jdk.tls.maxClientCertificateChainLength or jdk.tls.maxServerCertificateChainLength
+     * system property is not set, it will default to the value of the
+     * jdk.tls.maxCertificateChainLength system property. If either of them is
+     * set, it will supersede the value of the jdk.tls.maxCertificateChainLength
+     * system property.
+     */
+    static {
+        Integer clientLen = GetIntegerAction.privilegedGetProperty(
+                "jdk.tls.maxClientCertificateChainLength");
+        maxClientCertificateChainLength = (clientLen != null) ?
+                clientLen : maxCertificateChainLength;
+
+        Integer serverLen = GetIntegerAction.privilegedGetProperty(
+                "jdk.tls.maxServerCertificateChainLength");
+        maxServerCertificateChainLength = (serverLen != null) ?
+                serverLen : maxCertificateChainLength;
+    }
```

Comment:
I wonder if we should take the opportunity here with these new properties as well as `jdk.tls.maxCertificateChainLength` to also equate negative numbers (and maybe zero) to be the default.  Right now only property values that fail the internal parseInt conversion will evaluate to `null` and would be assigned the default I think.  But a negative value I think would be taken as-is from the property.  Should a negative max cert chain length get set to the default?  If so, it might also make sense to give a warning about the offending value and note that it is being set to the default (similar to what `GetPropertyAction.privilegedGetTimeoutProp()` does).
If you think this is worthwhile, the CSR should probably be updated to reflect that also.

Reply:
Change made to set to the default when a negative value is set for these system properties. Updated CSR for this.